dd-bridge-android/src/main/kotlin/com/datadog/android/bridge/internal/BridgeTrace.kt

```diff
@@ -7,20 +7,26 @@
 package com.datadog.android.bridge.internal
 
 import com.datadog.android.bridge.DdTrace
+import com.datadog.android.tracing.AndroidTracer
 import io.opentracing.Span
-import io.opentracing.util.GlobalTracer
+import io.opentracing.Tracer
 import java.util.concurrent.TimeUnit
 
 internal class BridgeTrace : DdTrace {
 
     private val spanMap: MutableMap<String, Span> = mutableMapOf()
+    // lazy here is on purpose. The thing is that this class will be instantiated even
+    // before Sdk.initialize is called, but Tracer can be created only after SDK is initialized.
+    private val tracer by lazy { tracerProvider.invoke() }
+    // this is needed for the test only, because Lazy cannot be settable
+    internal var tracerProvider: () -> Tracer = { AndroidTracer.Builder().build() }
```

Comment:
Instead of using a var, you could make it a constructor argument with a default value

Reply:
done, switched to the constructor injection.